packages/react-core/src/hooks/useSetUserAgent.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from 'react';
+import { setUserAgent, SetUserAgentOptions } from '@aws-amplify/ui';
+
+export default function useSetUserAgent({
+  componentName,
+  packageName,
+  version,
+}: SetUserAgentOptions): void {
+  useEffect(() => {
+    const clearUserAgent = setUserAgent({
+      componentName,
+      packageName,
+      version,
+    });
+
+    return () => {
+      clearUserAgent();
+    };
```

Comment:
Instead of wrapping the call to `clearUserAgent` in anonymous function can simplify the return to:
```suggestion
    return clearUserAgent;
```